### ISSUE TITLE
Setting password to None in URI without password adds '@'

### ISSUE THREAD

**Issue by hcarty** [mirage/ocaml-uri#78]
Example:
```ocaml
let uri = Uri.of_string "http://example.com/" in
Uri.with_password uri None
|> Uri.to_string
```
gives `http://@example.com/`

**Comment by dsheets** [comment]
That's indeed surprising behavior. I've fixed it in 5c6210a. I'm not sure what I was thinking when I wrote that test...

**Comment by hcarty** [comment]
Thanks for the quick fix!

**Comment by dsheets** [comment]
Now published by ocaml/opam-repository#4326 as `uri.1.9.1`.
